pkg/controllermanager/controller/shoot/shoot_conditions_control.go

```diff
@@ -141,3 +134,33 @@ func FilterSeedForShootConditions(obj, oldObj, _ client.Object, _ bool) bool {
 
 	return false
 }
+
+// MergeSeedConditions merges the conditions from the seed in the shoot avoiding duplications
+func MergeSeedConditions(seed *gardencorev1beta1.Seed, shoot *gardencorev1beta1.Shoot) []gardencorev1beta1.Condition {
+	seedConditions := []gardencorev1beta1.Condition{}
+	if seed != nil {
+		seedSystemComponentsCondition := gardencorev1beta1helper.GetOrInitCondition(seed.Status.Conditions, gardencorev1beta1.SeedSystemComponentsHealthy)
+		seedSystemComponentsConditionInShoot := gardencorev1beta1.Condition{
+			Type:               SeedSystemComponentsHealthyInShoot,
+			Status:             seedSystemComponentsCondition.Status,
+			LastTransitionTime: seedSystemComponentsCondition.LastTransitionTime,
+			LastUpdateTime:     seedSystemComponentsCondition.LastUpdateTime,
+			Message:            seedSystemComponentsCondition.Message,
+			Reason:             seedSystemComponentsCondition.Reason,
+			Codes:              seedSystemComponentsCondition.Codes,
+		}
+		seedConditions = append(gardencorev1beta1helper.RemoveConditions(seed.Status.Conditions, gardencorev1beta1.SeedSystemComponentsHealthy), seedSystemComponentsConditionInShoot)
+	}
+	seedConditionTypes := []gardencorev1beta1.ConditionType{
```

Comment:
You could keep this (line 154 to 161) at the start of the function and then only have one call to `if seed != nil` where you have the logic for modifying the `SeedSystemComponentsHealth` condition.